grpc/src/main/java/com/linecorp/armeria/client/grpc/GrpcClientBuilder.java

```diff
@@ -393,9 +394,13 @@ public <T> T build(Class<T> clientType) {
         if (!clientInterceptors.isEmpty()) {
             option(INTERCEPTORS.newValue(clientInterceptors));
         }
-        if (exceptionHandler != null) {
-            option(EXCEPTION_HANDLER.newValue(exceptionHandler));
+        if (exceptionHandler == null) {
+            exceptionHandler = GrpcExceptionHandlerFunction.ofDefault();
```

Comment:
I believe that we shouldn't set the property while building an instance:
```java
GrpcExceptionHandlerFunction exceptionHandler;
if (this.exceptionHandler == null) {
    exceptionHandler = GrpcExceptionHandlerFunction.ofDefault();
} else {
    exceptionHandler = this.exceptionHandler.orElse(GrpcExceptionHandlerFunction.ofDefault());
}
exceptionHandler = new UnwrappingGrpcExceptionHandleFunction(exceptionHandler);
option(EXCEPTION_HANDLER.newValue(exceptionHandler));
```

Reply:
Currently, exceptionHandler is passed through `ClientBuilderParams params` for instantiating `ArmeriaChannel`. This look like a current way to pass parameters by `ClientBuilderParams`. I may introduce exceptionHandler as new constructor parameter but I'm not sure that is right to do. Do you have any idea on this?

https://github.com/line/armeria/blob/8ab42847c146b481c72f000dbc1c4d77dc009220/grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaChannel.java#L104-L127

Reply:
The `exceptionHandler` is retrieved from `options` so we don't need to add `exceptionHandler` to the constructor parameter:
https://github.com/line/armeria/blob/8ab42847c146b481c72f000dbc1c4d77dc009220/grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaChannel.java#L127

I also realized that we have to create a new `exceptionHandler` only when it's set:
```java
public <T> T build(Class<T> clientType) {
    requireNonNull(clientType, "clientType");

    final List<ClientInterceptor> clientInterceptors = interceptors.build();
    if (!clientInterceptors.isEmpty()) {
        option(INTERCEPTORS.newValue(clientInterceptors));
    }
    if (exceptionHandler != null) {
        option(EXCEPTION_HANDLER.newValue(new UnwrappingGrpcExceptionHandleFunction(
                exceptionHandler.orElse(GrpcExceptionHandlerFunction.of()))));
    }
    ....
}

public static final ClientOption<GrpcExceptionHandlerFunction> EXCEPTION_HANDLER =
        ClientOption.define("EXCEPTION_HANDLER", new UnwrappingGrpcExceptionHandleFunction(
                GrpcExceptionHandlerFunction.of()));
               
```